api-gateway/src/main/kotlin/org/cqfn/save/gateway/security/WebSecurityConfig.kt

```diff
@@ -108,11 +135,34 @@ class WebSecurityConfig(
                 RedirectServerAuthenticationFailureHandler("/error")
             )
         }
+        .httpBasic { httpBasicSpec ->
+            // Authenticate by comparing received basic credentials with existing one from DB
+            httpBasicSpec.authenticationManager(
+                UserDetailsRepositoryReactiveAuthenticationManager(
+                    object : ReactiveUserDetailsService {
+                        override fun findByUsername(username: String): Mono<UserDetails> {
+                            val user: Mono<StringResponse> = webClient.get()
+                                .uri("/internal/users/$username")
```

Comment:
Identity source should be used here too, right? Do you now validate that the returned user has the same source as the caller?

Reply:
It's already passed from http client. `username` here is actually userInformation: `github@username`, in backend we will split this fields and call `findByUsernameAndSource`

Reply:
OK, good. Can you please make it more explicit here, at least by comment or better by explicit check?

Reply:
Added